src/Core/Types/Types/Relay/Extensions/RelayIdFieldExtensions.cs

```diff
@@ -176,21 +172,25 @@ public static class RelayIdFieldExtensions
             };
         }
 
-        private static FieldValueSerializer CreateSerializer(
+        private static IdFieldValueSerializer CreateIdFieldValueSerializer(
             ICompletionContext completionContext,
             Type type,
             string? typeName)
         {
-            var innerSerializer =
+            IIdSerializer innerSerializer =
                 (IIdSerializer)completionContext.Services.GetService(typeof(IIdSerializer)) ??
                 new IdSerializer();
 
-            return new FieldValueSerializer(
+            IIdFieldValueSerializerFactory idFieldValueSerializerFactory =
+                (IIdFieldValueSerializerFactory)completionContext.Services.GetService(typeof(IIdFieldValueSerializerFactory)) ??
+                new IdFieldValueSerializerFactory();
+
+            return idFieldValueSerializerFactory.Create(
                 typeName is { } ? typeName : completionContext.Type.Name.Value,
                 innerSerializer,
                 typeName is { },
                 DotNetTypeInfoFactory.IsListType(type),
-                DotNetTypeInfoFactory.GetInnerListType(type));
+                DotNetTypeInfoFactory.Unwrap(type));
```

Comment:
`GetInnerListType` would return `null` for non-list-wrapped argument types. I believe this is ok, but would want your OK on it. There's no use of the passed ValueType other than to create the list type, 
`ListType = CreateListType(valueType)`

That method falls through to be `string`, but potentially it'd be better if `ListType` was nullable as that'd be more semantic I guess. Will suggest that change in that file.